### ISSUE TITLE
Rendering tables crashe

### ISSUE THREAD

**Issue by sftse** [jugglerchris/rust-html2text#65]
Even after the fix in #64 some tables still crash, hitting the `assert!(width > 0)` in `text_renderer.rs`.

A reduced example is below
```
<!DOCTYPE html>
<html>
  <table><tbody><tr><td>
          <ol><li>ဘိန်းမုန့်</li></ol></td>
        <td>
          <ol><li>မုန့်ကြာစိ</li>
            <li>မုန့်တီ</li>
            <li>ခိုတောင်မုန့်တီ></li>
            <li>မုန့်ဟင်းခ</li>
            <li>တစ်ပင်တိုင်မုန့်ဟင်းခါး</li>
            <li>မုန့်စိမ်းပေါင်း</li>
            <li>မြိတ်ကတ်ကြေးကိုက်</li>
            <li>မြီးရှည်</li>
          </ol>
        </td>
        <td>
          <ol>
            <li></li></ol></td>
      </tr>
    </tbody>
  </table>
</html>
```
I have not further investigated and may not be able to for a while, but removing the assert condition leads to this example being processed reasonably, while processing the whole html this originated from appears to hit an infinite loop filling memory until OOM.

**Comment by jugglerchris** [comment]
Thanks for the report!  I think removing that assert seems reasonable, but there may be some issue exposed by that which is causing the infinite loop (e.g. trying to wrap text into a zero-width builder).  If you're able to come up with a case which triggers that, that would be amazing - but I'll see if I can experiment and find it in the meantime.

**Comment by sftse** [comment]
Yes, of course! I just wasn't initially sure I could recover the link of the page as it was obtained by a crawler.

[Example](https://my.wikipedia.org/w/index.php?title=%E1%80%99%E1%80%BC%E1%80%94%E1%80%BA%E1%80%99%E1%80%AC%E1%80%B7_%E1%80%A1%E1%80%85%E1%80%AC%E1%80%B8%E1%80%A1%E1%80%85%E1%80%AC)

**Comment by jugglerchris** [comment]
Thanks!  It was indeed trying to wrap into a zero-width builder.  The size calculation wasn't accounting for extra text generated for link targets.

**Comment by jugglerchris** [comment]
Assuming #66 fixes the issue for you, do you need it to be released soonish?

**Comment by sftse** [comment]
Ran into another [crash](https://my.wikipedia.org/w/index.php?title=%E1%80%9D%E1%80%AE%E1%80%80%E1%80%AE%E1%80%95%E1%80%AE%E1%80%B8%E1%80%92%E1%80%AE%E1%80%B8%E1%80%9A%E1%80%AC%E1%80%B8:%E1%80%85%E1%80%AE%E1%80%99%E1%80%B6%E1%80%81%E1%80%94%E1%80%B7%E1%80%BA%E1%80%81%E1%80%BD%E1%80%B2%E1%80%9E%E1%80%B0%E1%80%99%E1%80%BB%E1%80%AC%E1%80%B8)

**Comment by jugglerchris** [comment]
Ok, I've removed the asserts after all on #67 and that page renders for me now.
Thanks again!

**Comment by sftse** [comment]
Thanks, confirm that a crawl of 5.5k pages exhibits no crashes.

**Comment by jugglerchris** [comment]
Awesome!
I've just released 0.4.4 with the fixes.
